component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,409 @@
+    private List<SubThread> threadList = new ArrayList<>();
+    private static final String CURRENT_MAP_KEY = "current.map.key";
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public StateFactory<FileHandlerState> init(SourceEventListener sourceEventListener,
+                                               OptionHolder optionHolder,
+                                               String[] requiredProperties,
+                                               ConfigReader configReader,
+                                               SiddhiAppContext siddhiAppContext) throws SiddhiAppValidationException {
+        this.siddhiAppContext = siddhiAppContext;
+        this.sourceEventListener = sourceEventListener;
+        if (optionHolder.isOptionExists(Constants.URI)) {
+            uri = optionHolder.validateAndGetStaticValue(Constants.URI);
+            FileObject file = Utils.getFileObject(uri);
+            try {
+                if (!file.exists()) {
+                    throw new SiddhiAppCreationException(" Directory/File " + file.getPublicURIString()
+                            + " is not found ");
+                }
+                if (file.isFile()) {
+                    listeningFileUri = file.getName().getPath();
+                    file = file.getParent();
+                }
+                listeningDirUri = file.getName().getPath();
+            } catch (FileSystemException e) {
+                throw new SiddhiAppValidationException("Directory/File " + file.getPublicURIString()
+                        + " is not found.", e);
+            }
+        }
+        if (uri == null) {
+            throw new SiddhiAppCreationException("uri is not found.");
+        }
+        String monitoringValue = optionHolder.validateAndGetStaticValue(Constants.MONITORING_INTERVAL, "500");
+        try {
+            monitoringInterval = Long.parseLong(monitoringValue);
+        } catch (NumberFormatException e) {
+            throw new SiddhiAppRuntimeException("Value provided for monitoring " + monitoringValue +
+                    " is invalid.", e);
+        }
+        return FileHandlerState::new;
+    }
+
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[]{String.class, byte[].class};
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback,
+                        FileHandler.FileHandlerState fileHandlerState) {
+        initiateFileAlterationObserver();
+        File[] listOfFiles = new File(listeningDirUri).listFiles();
+        if (listOfFiles != null) {
+            for (File listOfFile : listOfFiles) {
+                initialMap.put(listOfFile.getName(), listOfFile.length());
+                sourceEventListener.onEvent(getFileHandlerEvent(listOfFile, listeningFileUri,
+                        STATUS_DONE), null);
+            }
+        }
```

Comment:
This is not necessary. We need to compare the 'listOfFiles' with the map we receive from the 'restore' method and generate file events accordingly

Reply:
fixed in 3f9a2a0